### PR TITLE
[SPARK-48786] Define Config Loading Framework for Spark Operator Controller

### PR DESCRIPTION
**Pull request by jiangzho**
### What changes were proposed in this pull request?

This is a breakdown PR of #12 - introducing config loading framework for controller module. It includes a config map reconciler that can watch properties updates for hot properties loading.

### Why are the changes needed?

These are needed to enable user-configurable behavior of the operator.

### Does this PR introduce _any_ user-facing change?

No to existing Spark users.

### How was this patch tested?

Tested via unit test coverage. 

### Was this patch authored or co-authored using generative AI tooling?

No.

### CODE REVIEW

File: spark-operator/src/main/resources/EcsLayout.json [hunk 1]
```diff
@@ -0,0 +1,49 @@
+{
```
Comment (dongjoon-hyun) [inline spark-operator/src/main/resources/EcsLayout.json:1]: Shall we move this to `test/resources`?

File: spark-operator/src/main/java/org/apache/spark/k8s/operator/config/SparkOperatorConf.java [hunk 2]
```diff
@@ -0,0 +1,469 @@
...
+
+import org.apache.spark.k8s.operator.Constants;
+import org.apache.spark.k8s.operator.SparkApplication;
+
+/** Spark Operator Configuration options. */
+@Slf4j
+public class SparkOperatorConf {
+  public static final ConfigOption<String> OperatorAppName =
+      ConfigOption.<String>builder()
+          .key("spark.kubernetes.operator.name")
```
Comment (dongjoon-hyun) [inline spark-operator/src/main/java/org/apache/spark/k8s/operator/config/SparkOperatorConf.java:45]: Please remove all actual configuration (`spark.kubernetes.operator.*`) variables from this PR in order to focus on `ConfigOption` framework.

Comment (dongjoon-hyun) [reply spark-operator/src/main/java/org/apache/spark/k8s/operator/config/SparkOperatorConf.java:45]: Specifically, let's remove from line 43 to 405 and all dependent lines.

**Review by dongjoon-hyun — CHANGES_REQUESTED** [review]
Thank you for making a narrow-downed PR. We need to focus on new `ConfigOption` framework itself in this PR, @jiangzho . Could you address the following?
- Please remove all actual configuration (spark.kubernetes.operator.*) variables from this PR. (https://github.com/apache/spark-kubernetes-operator/pull/17/files#r1664519696)
- Let's move `EcsLayout.json` to `test/resources`. (https://github.com/apache/spark-kubernetes-operator/pull/17/files#r1664512208)
- Also, please revise the PR title accordingly.

File: spark-operator/src/main/java/org/apache/spark/k8s/operator/config/SparkOperatorConf.java (hunk 2 above)
Comment (jiangzho) [reply spark-operator/src/main/java/org/apache/spark/k8s/operator/config/SparkOperatorConf.java]: sure - this PR has been narrowed down to config loaders only

File: spark-operator-api/src/main/java/org/apache/spark/k8s/operator/Constants.java [hunk 3]
```diff
@@ -26,6 +26,7 @@ public class Constants {
   public static final String LABEL_SPARK_OPERATOR_NAME = "spark.operator/name";
   public static final String LABEL_SENTINEL_RESOURCE = "spark.operator/sentinel";
   public static final String LABEL_RESOURCE_NAME = "app.kubernetes.io/name";
+  public static final String LABEL_COMPONENT_NAME = "app.kubernetes.io/component";
```
Comment (dongjoon-hyun) [inline spark-operator-api/src/main/java/org/apache/spark/k8s/operator/Constants.java:29]: Reordering this?

**Comment by dongjoon-hyun** [comment]
Could you make CI happy, @jiangzho ?

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM (Pending CIs).

**Pull request closed without merging**